### PR TITLE
mine

### PR DESCRIPTION
**Pull request by Menna120**
give me your openion in what i have done

### CODE REVIEW

**Comment by 3105731** [comment]
jghcgx

File: src/main/kotlin/interactor/Consumption.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+         }
+        return x.toSortedMap()
+            .values
+            .last()
+            .toString()
+    }
+}
+private fun excludeNullValue(city: CityEntity): Boolean {
+    return city.averageMonthlyNetSalaryAfterTax != null
+            && city.dataQuality
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt]: He didn't mention to check the dataQuality in task 10, so you shouldn't exclude values according to dataQuality

File: src/main/kotlin/interactor/Consumption.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package interactor
+
+import model.CityEntity
+
+
+class Consumption (private val dataSource: CostOfLivingDataSource){
+
+    fun execute(): String {
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt]: function should return CityEntity not the name of the city as mentioned

File: src/main/kotlin/interactor/Consumption.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+private fun excludeNullValue(city: CityEntity): Boolean {
+    return city.averageMonthlyNetSalaryAfterTax != null
+            && city.dataQuality
+            &&city.realEstatesPrices.apartment3BedroomsOutsideOfCentre != null
+            &&city.foodPrices.riceWhite1kg != null
+            &&city.foodPrices.chickenFillets1kg != null
+            &&city.foodPrices.beefRound1kgOrEquivalentBackLegRedMeat != null
+            &&city.foodPrices.localCheese1kg != null
+            &&city.foodPrices.loafOfFreshWhiteBread500g != null
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt]: line 37, 38, 39, 40
he didn't mention any of this I know that is also from consumptions, but you should be strict with which asked to check in task

File: src/main/kotlin/interactor/Consumption.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            &&city.servicesPrices.basicElectricityHeatingCoolingWaterGarbageFor85m2Apartment != null
+            &&city.servicesPrices.internet60MbpsOrMoreUnlimitedDataCableAdsl != null
+            &&city.servicesPrices.internationalPrimarySchoolYearlyForOneChild != null
+            &&city.servicesPrices.oneMinOfPrepaidMobileTariffLocalNoDiscountsOrPlans != null
+}
+
+private fun servicesCoast(city:CityEntity):Float{
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt]: servicesCost function all calculations is not mentioned in problem.

File: src/main/kotlin/interactor/Consumption.kt [hunk 5]
```diff
@@ -5,58 +5,39 @@ import model.CityEntity
...
-            .filter(::excludeNullValue)
-            .sortedBy{it.realEstatesPrices.apartment3BedroomsOutsideOfCentre}
-        cities.forEach {
-             val servicesCoast= servicesCoast(it)
-             val remainder= remainder(it,servicesCoast)
-             if(isSuitable(servicesCoast,remainder)){
-                 if (x.contains(remainder)) x[remainder]?.add(it.cityName)
-                 else x[remainder] = mutableListOf(it.cityName)
+            .filter(::excludeNullOrZeroValue)
+            .forEach {
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt:13]: line 13 to the function return in line 20
try to make the code more cleaner and readable think in another way you can do the same thing

File: src/main/kotlin/interactor/Consumption.kt [hunk 6]
```diff
@@ -5,58 +5,39 @@ import model.CityEntity
...
-    return (city.servicesPrices.basicElectricityHeatingCoolingWaterGarbageFor85m2Apartment!!
-        +(city.servicesPrices.internet60MbpsOrMoreUnlimitedDataCableAdsl!!)
-        +(city.servicesPrices.internationalPrimarySchoolYearlyForOneChild!!/6)
-        +(city.servicesPrices.oneMinOfPrepaidMobileTariffLocalNoDiscountsOrPlans!!))
-    }
-private fun remainder(city:CityEntity,servicesCoast:Float):Float{
+private fun remainder(city:CityEntity):Float{
 
     return( city.averageMonthlyNetSalaryAfterTax!!*2
             -city.realEstatesPrices.apartment3BedroomsOutsideOfCentre!!
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/interactor/Consumption.kt:39]: line 37 you have apartments that inside the city and outside the city you should get the minimum amount between them it is just a logic doesn't mean something to me just to make you aware only..

File: src/main/kotlin/Main.kt [hunk 7]
```diff
@@ -1,20 +1,13 @@
...
-import interactor.GetCityHasCheapestInternetConnectionInteractor
-import interactor.GetHighestSalaryAverageCititesNamesInteractor
-
 fun main() {
     val csvParser = CsvParser()
     val dataSource: CostOfLivingDataSource = CsvDataSource(csvParser)
 
-    val getHighestSalaryAverageCities = GetHighestSalaryAverageCititesNamesInteractor(dataSource)
-    println(getHighestSalaryAverageCities.execute(limit = 10))
+    print(Consumption(dataSource).execute())
```
Comment (ZeyadAbdullah679) [inline src/main/kotlin/Main.kt:9]: You did it Annapolis is right!

Congratulations 👏

**Review by ZeyadAbdullah679 — CHANGES_REQUESTED** [review]
You did well, now try to clean your code using only equal operator for return in execute() do your best!
I mean like this fun execute()=something

**Pull request closed without merging**